access/templates/access/graded_form.html

```diff
@@ -1,32 +1,39 @@
 {% load i18n %}
-<form method="post" action="{{ post_url }}"{% if result.form.multipart %} enctype="multipart/form-data"{% endif %}>
+{% load access %}
+<form method="post" action="{{ post_url }}"{% if result.form.multipart %} enctype="multipart/form-data"{% endif %} class="aplus-quiz1">
 	{% for field in result.form %}
+
 	{% if field.field.open_set %}
-	<fieldset id="{{ field.field.open_set }}"{% if result.accepted and field.field.group_errors %} class="alert {% if field.field.open_set in result.error_groups %}alert-danger{% else %}alert-success{% endif %}"{% endif %}>
+	<!-- Beginning of all questions -->
+		<fieldset id="{{ field.field.open_set }}"{% if result.accepted and field.field.group_errors %} class="alert {% if field.field.open_set in result.error_groups %}alert-danger{% else %}alert-success{% endif %}"{% endif %}>
 		{% if field.field.set_title %}<legend>{{ field.field.set_title }}</legend>{% endif %}
 	{% endif %}
 
 		{% if field.field.type == "static" %}
-		{{ field.field.more|safe }}
+		<div class="question-description">
+			{{ field.field.more|safe }}
+		</div>
 		{% elif field.field.type == "hidden" %}
 			{{ field }}
 		{% else %}
 
 		{% if not field.field.row_label or field.field.open_table %}
-		<div class="{{ field.field.html_class }} {% if field.errors %} has-error{% elif not field.field.group_errors and field.field.name in result.error_fields %} has-error{% endif %}">
+		<!-- Individual question heading (individual form-group) -->
+		<div class="question-area {{ field.field.html_class }} {% if field.errors %} has-error{% endif %}">
 
 			{% if field.field.required %}<span class="required">*</span>{% endif %}
 
 			{% if field.label %}
-			{% if field.field.choice_list %}
-			<label>{{ field.label }}</label>
-			{% else %}
-			{{ field.label_tag }}
-			{% endif %}
+				{% if field.field.choice_list %}
+					<label class="control-label">{{ field.label }}</label>
```

Comment:
It is for bootstrap. When there is a validation error with that field, the label is bolded and colored red, to indicate the field with the error. I tried and it doesn't work without that control-label. I guess bootstrap recognizes the parts that go together, since they are any case within the same form-group. And that too has the auto_id improvement coming here :) https://github.com/apluslms/mooc-grader/pull/78

Reply:
I was thinking of adding `for="{{ field.id_for_label }}"` to the label, however I'm not sure if it's missing for a reason.

Reply:
Based on a very quick look, labels have the `for` attribute on text fields and not on these multi-value choice fields (checkboxes, radio buttons). That actually makes sense, right? If you have five checkboxes, clicking the question label above all of them should not tick any one of the checkboxes. That would be different for labels tied to one specific checkbox.

Reply:
Ah, that was the thing I was missing! Now it makes sense again. Carry on.